atoma-service/src/streamer.rs

```diff
@@ -311,48 +319,65 @@ impl Streamer {
             total_compute_units,
         );
 
-        // Calculate and update total hash
-        let total_hash = blake2b_hash(&[self.payload_hash, response_hash].concat());
-        let total_hash_bytes: [u8; 32] = total_hash
-            .as_slice()
-            .try_into()
-            .expect("Invalid BLAKE2b hash length");
-
-        // Update stack total hash
-        if let Err(e) =
-            self.state_manager_sender
-                .send(AtomaAtomaStateManagerEvent::UpdateStackTotalHash {
-                    stack_small_id: self.stack_small_id,
-                    total_hash: total_hash_bytes,
-                })
-        {
-            error!(
-                target = "atoma-service-streamer",
-                level = "error",
-                endpoint = self.endpoint,
-                "Error updating stack total hash: {}",
-                e
-            );
-        }
+        if let Some(stack_small_id) = self.stack_small_id {
+            // Calculate and update total hash
+            let total_hash = blake2b_hash(&[self.payload_hash, response_hash].concat());
+            let total_hash_bytes: [u8; 32] = total_hash
+                .as_slice()
+                .try_into()
+                .expect("Invalid BLAKE2b hash length");
+
+            // Update stack total hash
+            if let Err(e) =
+                self.state_manager_sender
+                    .send(AtomaAtomaStateManagerEvent::UpdateStackTotalHash {
+                        stack_small_id,
+                        total_hash: total_hash_bytes,
+                    })
+            {
+                error!(
+                    target = "atoma-service-streamer",
+                    level = "error",
+                    endpoint = self.endpoint,
+                    "Error updating stack total hash: {}",
+                    e
+                );
+            }
 
-        // Update stack num tokens
-        let concurrent_requests = handle_concurrent_requests_count_decrement(
-            &self.concurrent_requests,
-            self.stack_small_id,
-            &self.endpoint,
-        );
-        if let Err(e) = update_stack_num_compute_units(
+            // Update stack num tokens
+            let concurrent_requests = handle_concurrent_requests_count_decrement(
+                &self.concurrent_requests,
+                stack_small_id,
+                &self.endpoint,
+            );
+            if let Err(e) = update_stack_num_compute_units(
```

Comment:
Do we want to unlock possible locked payments if something fails ?

Reply:
in reality the locked payments don't have to be there, the locking was introduced so we don't overshoot the stack. But never happens to fiat. But I've added this to see if there is inference running and for how much. It's just an informative column.